internal/contentmanager/contentmanager_test.go

```diff
@@ -108,6 +108,48 @@ func TestWatch(t *testing.T) {
 	}
 }
 
+func BenchmarkWatch(b *testing.B) {
+	rcm := func(_ context.Context, _ client.Object, cfg *rest.Config) (*rest.Config, error) {
+		return cfg, nil
+	}
+	config := &rest.Config{}
+
+	ce := &v1alpha1.ClusterExtension{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster-extension",
+		},
+	}
+	objs := []client.Object{
+		&corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "webserver",
+			},
+		},
+	}
+
+	for n := 0; n < b.N; n++ {
+		mgr, _ := manager.New(config, manager.Options{})
+		ctrl, err := controller.New("test-controller", mgr, controller.Options{
+			Reconciler: reconcile.Func(func(context.Context, reconcile.Request) (reconcile.Result, error) {
+				return reconcile.Result{}, nil
+			}),
+		})
+		if err != nil {
+			b.Logf("creating controller; %v", err)
+		}
+
+		instance := New(rcm, config, mgr.GetRESTMapper())
```

Comment:
This is quite a bit of setup that will impact the benchmark results, and I would expect that quite a bit CPU time and memory allocation happens in this setup, potentially much more than in the `instance.Watch` call.

Reply:
Removing initial tests and focusing on areas more likely to be impactful on performance